common/djangoapps/student/models.py

```diff
@@ -1328,6 +1330,56 @@ def is_paid_course(self):
 
         return False
 
+    def is_active_course(self, cert_status):
+        """ Returns True if a course is active.
+
+        A course is considered active if its end date is in future.
+
+        Args:
+            cert_status (Dict): Certificate status for this enrollment
+
+        """
+        if self.course_overview.end is None:
+            return True
+
+        now = timezone.now()
+        if self.course_overview.end > now:
+            return True
+
+        if self.course_overview.end < now and cert_status.get('status') == 'processing':
+            return True
+
+        return False
+
+    def is_completed_course(self, cert_status):
```

Comment:
Same as above. Why does certificate status affect course status?

Would it be simpler to have have one method: `get_course_status`, perhaps on the `CourseOverview` class?